src/StoreMetaUnpacker.h

```diff
@@ -9,36 +9,56 @@
 #ifndef SQUID_TYPELENGTHVALUEUNPACKER_H
 #define SQUID_TYPELENGTHVALUEUNPACKER_H
 
-class StoreMeta;
-class StoreEntry;
+#include "StoreMeta.h"
+
+namespace Store {
+
+/// iterates swapped out swap meta fields, loaded into a given buffer
+class SwapMetaIterator
+{
+public:
+    /* some of the standard iterator traits */
+    using iterator_category = std::forward_iterator_tag;
+    using value_type = const SwapMetaView;
+    using pointer = value_type *;
+    using reference = value_type &;
+
+    /// positions iterator at the start of a swap meta field extending up to end
+    SwapMetaIterator(const void *start, const void *end);
+
+    /* some of the standard iterator methods */
+    reference operator *() const { return meta_; }
+    pointer operator ->() const { return &meta_; }
+    SwapMetaIterator& operator++();
+    bool operator ==(const SwapMetaIterator &them) const { return fieldStart_ == them.fieldStart_; }
+    bool operator !=(const SwapMetaIterator &them) const { return !(*this == them); }
+
+private:
+    void sync();
+
+    const char *fieldStart_; ///< the start of the current field
+    const void * const bufEnd_; ///< last field must end at this boundary
+    SwapMetaView meta_; ///< current field; valid after sync() and before end
+};
+
+} // namespace Store
 
 class StoreMetaUnpacker
 {
 
 public:
     StoreMetaUnpacker (const char *buf, ssize_t bufferLength, int *hdrlen);
-    StoreMeta *createStoreMeta();
-    bool isBufferZero(); ///< all-zeros buffer, checkBuffer() would throw
-    /// validates buffer sanity and throws if validation fails
-    void checkBuffer();
+
+    // for-range loop API for iterating over serialized swap metadata fields
+    using Iterator = Store::SwapMetaIterator;
+    Iterator cbegin() const { return Iterator(metas, metas + metasSize); }
+    Iterator cend() const { return Iterator(metas + metasSize, metas + metasSize); }
```

Comment:
AFAICT, the intent behind our (only) iterator _is_ a constant iterator  -- our iterator does not modify the container being iterated in any way. So, in a sense, we only want to provide cbegin/cend(), but we cannot do that, IIRC, because for-range loop iteration use cases require begin/end(). Thus, we provide both, enabling both the existing use case (for-range loops) and possible future use cases (explicit constant iteration).

If we should do more for our existing iterator to actually be a "constant iterator" from C++/STL point of view as well, we should do that! I do not know what that entails.